### ISSUE TITLE
Warnings about name collision of modules

### ISSUE THREAD

**Issue by osch** [luarocks/luarocks#1148]
Could it be possible to have a warning message when installing a rock that contains a module that is already provided by another rock? Or is there already an option for this and I didn't find it?

At least with Luarocks version 3.2.1 I verified that there are no warnings for this case. 

However it seems that Luarocks is aware of the collision and installs the new module under another name and activates the, if the rock containing the first module is deinstalled.

**Comment by hishamhm** [comment]
That's a good idea! Indeed LuaRocks performs this detection internally, it would not be hard to make it print a message when a conflict is detected.

**Comment by npalladium** [comment]
Is this still open? If yes, I'd be interested in taking it up. (Might take some time as I'm unfamiliar with LuaRocks internals).

**Comment by hishamhm** [comment]
@npalladium Yes, this is still open! That logic happens in the `luarocks.repos` module (that logic is pretty involved, but the conflict detection happens in `repos.deploy_files` deciding whether to install or to rename the file (when the existing file is not a LuaRocks-installed file, it "backs it up" it with a `~` suffix instead).

**Comment by npalladium** [comment]
Just to make sure that I have understood it correctly: 
- The modules I have to be concerned with are `luarocks.cmd.install` & `luarocks.repos` (and maybe `luarocks.remove` & `luarocks.purge`). 
- I will have to use `luarocks.util.printout` or `luarocks.util.printerr`. 

As this is a fairly small change, I'll probably add another comment to this issue soon about what I intend to do and as I'm a first time contributor, I will open a draft PR. Please let me know if this works.

**Comment by hishamhm** [comment]
@npalladium Sounds like a plan! Yes, the backend logic is in `luarocks.repos`, which is triggered by those other modules.

**Comment by sharadxsingh** [comment]
is this still open i would like to work on this issue i'm new here willing to contribute and learn

**Comment by hishamhm** [comment]
Yes, it is. PRs are welcome!
